enhancements/plain-k8s-bundles.md

```diff
@@ -0,0 +1,220 @@
+
+### New Mediatype
+
+The `k8s+v1` mediatype will be used to represent a bundle composed of plain Kubernetes manifests. This will let OLM and other tooling differentiate this type of bundle from other types.
+
+### Inferring CSV Content
+
+CSVs are generally ⅔s standard k8s manifests and ⅓ operator runtime information. They provide a description that can be used to gather dependencies for, install, and display an operator. When attempting to denormalize this description, it's critical that there is no loss of information capacity. To this end, it helps to split it into two categories: inferrable and non-inferrable.
+
+#### Inferrable
+
+To support a wider range of inferred CSV fields, manifests for resources usually generated by a CSV will need to be provided as input to `k8s+v1` bundle creation:
+
+- Deployments for InstallStrategy
+- RBAC w/ subjects matching Deployment ServiceAccounts for Permissions and Owned, Required, and NativeAPIs
+- APIServices and Services for APIServiceDefinitions
+
+Required, provided, optional, and native APIs in a `k8s+v1` bundle will inferred by inspecting the RBAC, CRDs, Deployments, and APIServices it contains using the following rules:
+
+- the GVK of any CRD in the bundle will be considered a **provided API**
+- a GVK referenced in RBAC with a subject shared by the Deployment will be considered:
+  - **optional** if the referencing `(Cluster)RoleBinding` has the `operators.operatorframework.io.bundle.dependency="optional"` annotation set
+    - i.e. the GVK is neither provided or required
+  - **provided** if it belongs to the group version of an APIService in the bundle
+  - **required** if it DOES NOT belong to:
+    - any group version defined by an APIService or CRD in the bundle
+    - a `*.k8s.io` group
+- `nonResourceURLs` referenced in RBAC with a subject shared by the Deployment will be considered **native APIs**
+
+The dependence on explicit kinds in RBAC rules means that rules with wildcard types (i.e. `*`) are disallowed in `k8s+v1` bundles. This may seem like a burdensome restriction, but a nice corollary is that all `k8s+v1` bundles follow [the principle of least privilege](https://en.wikipedia.org/wiki/Principle_of_least_privilege). The exception to the restriction is RBAC w/ subjects that do not match a ServiceAccount in the bundle's Deployments.
+
+#### Non-Inferrable
+
+The following are parts of a CSV that are NOT directly inferrable from plain kubernetes resources:
+
+- Name
+- Descriptors
+- Description
+- Display Name
+- Keywords
+- Labels
+- Links
+- Maintainers
+- Maturity
+- Provider
+- Selector
+- Icon
+- Runtime Metadata
+- MinKubeVersion
+- (Future) Optional APIs
+- Replaces
+- Version
+- InstallModes
+
+This non-inferrable data will be gathered through a mixture of resource annotations and an `olm.yaml` file in the `metadata` directory of the bundle.
+
+Ex. _olm.yaml_
+
+```yaml
+name: packageserver-v1.14.1
+maturity: alpha
+version: 0.14.1
+replaces: packageserver-v1.14.0
```

Comment:
I suspect that the audience of operator authors that migrate away from maintaining a CSV will also prefer to adopt the semver-based update graph. Should the structure of this `olm.yaml` file be versioned as an escape hatch for increasing complexity in these files?

Reply:
The entire bundle is versioned by mediatype. I removed replaces with that assumption -- I should add that it IS an assumption somewhere though.